aiogram/api/client/session/aiohttp.py

```diff
@@ -1,29 +1,72 @@
 from __future__ import annotations
 
-from typing import AsyncGenerator, Callable, Optional, TypeVar, cast
+from typing import AsyncGenerator, Callable, Optional, TypeVar, Tuple, Dict, Any, Union, cast
 
-from aiohttp import ClientSession, ClientTimeout, FormData
+from aiohttp import ClientSession, ClientTimeout, FormData, BasicAuth, TCPConnector
 
 from aiogram.api.methods import Request, TelegramMethod
 
 from .base import PRODUCTION, BaseSession, TelegramAPIServer
 
 T = TypeVar("T")
+_ProxyType = Union[str, Tuple[str, BasicAuth]]
 
 
-class AiohttpSession(BaseSession):
+class AiohttpSession(BaseSession[_ProxyType]):
     def __init__(
         self,
         api: TelegramAPIServer = PRODUCTION,
         json_loads: Optional[Callable] = None,
         json_dumps: Optional[Callable] = None,
+        proxy: Optional[_ProxyType] = None,
     ):
-        super(AiohttpSession, self).__init__(api=api, json_loads=json_loads, json_dumps=json_dumps)
+        super(AiohttpSession, self).__init__(
+            api=api,
+            json_loads=json_loads,
+            json_dumps=json_dumps,
+            proxy=proxy
+        )
         self._session: Optional[ClientSession] = None
+        self.cfg.connector_type = TCPConnector
+        self.cfg.connector_init = cast(Dict[str, Any], {})
+
+        if self.proxy:
+            try:
```

Comment:
What about HTTP proxy?
aiohttp already supports HTTP proxy and in this case external library is not needed

Reply:
library supports it too. I think, we shouldn't care about proxy. Aiohttp does not support socks proxies, AFAIK.